lib/ansible/executor/task_queue_manager.py

```diff
@@ -256,6 +257,15 @@ def cleanup(self):
 
     def _cleanup_processes(self):
         if hasattr(self, '_workers'):
+            for attempts_remaining in range(C.WORKER_SHUTDOWN_POLL_COUNT - 1, -1, -1):
```

Comment:
This feels odd to me - if I set the poll count to 1, I'd expect it to sleep (at most) once. With the `- 1` here, we enter the loop once, but the only value in the `range` is `0`, so when we get to the `if attempts_remaining` below, `attempts_remaining` is 0 (falsey) so we never sleep.

```suggestion
            for attempts_remaining in range(C.WORKER_SHUTDOWN_POLL_COUNT, -1, -1):
```

Reply:
This is intentional. I was trying to do a few things:

- Make it possible to enable the warning without adding a sleep.
- Avoid sleeping on the final iteration, so that the termination pass happens immediately.

I  agree the behavior could be unintuitive and should be commented. I think the documentation of the config at least accurately describes the behavior.